src/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java

```diff
@@ -242,6 +252,19 @@ private FormData prepareFormDataSet(FormFields form) {
 				currentList = submitFields;
 			}
 			for (String value : getValues(field)) {
```

Comment:
@psiinon @thc202  (comment for lines 255-267)
@cdlaguna and I have set it so that fields that contain the attribute 'value' are now passed to the valueGenerator. They will pass in the value associated as a parameter (defaultValue) and the DefaultValueGenerator will return the same 'defaultValue'. If a new value is provided by the valueGenerator to replace the default then it will be updated then passed to the Spider. We're open to feedback on this approach to handling predefined values.